### ISSUE TITLE
Some Header datasets in UVH5 memo lack units etc

### ISSUE THREAD

**Issue by david-macmahon** [RadioAstronomySoftwareGroup/pyuvdata#922]
The UVH5 memo specifies a number of `Header` datasets for a UVH5 file.  The datatypes and dataspaces (dimensions) of these datasets is well defined in the memo, but some of them lack information regarding what units should be used and occasionally other details.  Here is a list (that ended up being longer than I thought it would be when I started):

1. `phase_type`: Needs more clarity.  The two specified values are `phased` and `drift`, but it's not clear what `phased` means.  Is it that the telescope has physically tracked the phase center for the duration of the observation or that the fringe phase has been "stopped"?  This feels like a field that has been specified for fixed antennas but needs more fleshing out for steerable antennas.
1. `uvw_array`: Lack units. It has no `Nfreqs` dimension it can't be wavelengths, but is it meters or nanoseconds or ???
1. `antenna_positions`: Lacks units.  The frame is specified as "relative to the _telescope location_ (in the ITRF frame)", but "_telescope location_", despite being in italics, is not defined elsewhere other than perhaps the `latitude`, `longitude`, `altitude` (LLA) fields.  The sample file `zen.2458432.34569` has antenna positions such as `(5488676, -105, -3240024)`, which seems like ECEF coordinates in meters that have been rotated in longitude to the longitude of the LLA. The "LLA to ECEF" reference should also be specified (WGS84?).
1. `dut1`: Lacks units.  Most likely it's seconds, but should be explicitly stated as such for completeness.
1. `earth_omega`: Why degrees (per day) when all other angles are radians (other than `latitude`/`longitude` which are specified in degrees so commonly that it's tolerable to store them as degrees in the data file)?  I'm not advocating a change to this, but just pointing out that it would be nice to give some historical context for this design decision.
1. `gst0`: Lacks units.  Many possibilities come to mind, no "obvious" default unit for this (IMHO). Also specified as "Greenwich sidereal time at midnight on reference date", but is that Greenwich or "local" midnight?  Not clear whether this is apparent or mean sidereal time.  [Why is this even here?  Isn't this derivable from e.g. `time_array` values and `dut1`?]
1. `rdate`: This string field lacks format/resolution details.  Should be more clearly specified (even if its specified as a free-form "you figure it out" format).
1. `x_orientation`: This is already being discussed in #854, but free form text seems suboptimal IMHO.
1. `uvplane_reference_time`: Lacks units and origin (and frame if you want to get relativistic)
1. `phase_center_frame`: Not clear what the two specified values mean (`gcrs` and `icrs`).  Also not clear how `uvw_array` can be "phased to" a frame.  Isn't `(u,v,w)` just `B dot S` irrespective of any phasing that's been applied to the data?
1. `lst_array`: A brief explanation of the benefits of including this metadata-that-can-be-derived-from-other-metadata would be useful IMHO.
1. `extra_keywords`: This Header "subgroup" specifies that it is for the storage of "extra" (presumably non-required).  It says that this group contains datasets whose names are the "extra keywords" and that the extra values "are expected to be scalars, but UVH5 makes no formal restriction on this".  I think it would be good to specify further that they be comprised of HDF5 datatypes when feasible (e.g. as opposed to Python or Numpy "pickles", as is the case for sample file `zen.2458432.34569`).

**Comment by plaplant** [comment]
@david-macmahon I'm glad you find UVH5 useful! These are all good points, and we should have them in the UVH5 memo. I was planning on updating the memo as part of #912, so I can make sure these issues are addressed as well. I will let you know when I've made a PR on it, if you don't mind taking a look to make sure there's enough detail to answer these questions 🙂

**Comment by david-macmahon** [comment]
Sure thing!  I'd be happy to review any changes/PRs to the memo.  BTW, I also had questions about the spectral window stuff, but I saw #912 and decided to wait for that and only ask any questions that still remain.

**Comment by david-macmahon** [comment]
After looking at the `antenna_positions` datasets of all the UVH5 files in `pyuvdata/data`, I realized that the one file I had looked into previously (`zen.2458432.34569.uvh5`) uses a different coordinate frame than all the other UVH5 files (just my luck).  All of the other UVH5 files in the `pyuvdata` repo have antenna positions that appear to be, as the memo describes, in an (X,Y,Z) coordinate frame with the same basis vectors as the Earth Centered Earth Fixed (ECEF) International Terrestrial Reference Frame (ITRF), but with the origin translated from Earth Center to the "telescope location", which is presumably the location specified by `latitude`, `longitude`, and `altitude`.  Not sure whether the UVH5 spec needs to mandate the reference ellipsoid used for converting between LLA and ECEF or leave that as a free parameter, but either way it would still be good to discuss that fact in the memo.  I also think it would be great to have an appendix describing the conversion between the relative ITRF frame (XYZ) used for UVH5 antenna positions and the common (East, North, Up), aka ENU, relative topocentric frame used for antenna positions, but maybe that belongs more in the phasing memo than the UVH5 spec.

Not sure where the oddball antenna positions in the `zen.2458432.34569.uvh5` file came from, but that's probably worth its own issue.
